src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -19,148 +37,453 @@ private FluidUtil()
     {
     }
 
-    /** Returns true if interaction was successful. */
-    public static boolean interactWithTank(ItemStack stack, EntityPlayer player, IFluidHandler tank, EnumFacing side)
+    /**
+     * Used to handle the common case of a fluid item right-clicking on a fluid handler.
+     * First it tries to fill the container item from the fluid handler,
+     * if that action fails then it tries to drain the container item into the fluid handler.
+     *
+     * Returns true if interaction was successful.
+     * Returns false if interaction failed.
+     */
+    public static boolean interactWithFluidHandler(ItemStack stack, IFluidHandler fluidHandler, EntityPlayer player)
```

Comment:
This doesn't seem to handle creative mode correctly, where you'd not want the container to be consumed.